src/main/kotlin/com/sipios/springsearch/SpecificationImpl.kt

```diff
@@ -17,24 +19,24 @@ import java.util.ArrayList
  * @param <T> THe class on which the specification will be applied
 </T> */
 class SpecificationImpl<T>(private val criteria: SearchCriteria) : Specification<T> {
+    @Throws(ResponseStatusException::class)
     override fun toPredicate(root: Root<T>, query: CriteriaQuery<*>, builder: CriteriaBuilder): Predicate? {
         val nestedKey = criteria.key.split(".")
         val nestedRoot = getNestedRoot(root, nestedKey)
         val criteriaKey = nestedKey[nestedKey.size - 1]
-
-        when (criteria.operation) {
-            SearchOperation.EQUALS -> return builder.equal(nestedRoot.get<String>(criteriaKey), criteria.value)
-            SearchOperation.NOT_EQUALS -> return builder.notEqual(nestedRoot.get<String>(criteriaKey), criteria.value)
-            SearchOperation.GREATER_THAN -> return builder.greaterThan(nestedRoot.get<Double>(criteriaKey), criteria.value!!.toDouble())
-            SearchOperation.LESS_THAN -> return builder.lessThan(nestedRoot.get<Double>(criteriaKey), criteria.value!!.toDouble())
-            SearchOperation.STARTS_WITH -> return builder.like(nestedRoot.get(criteriaKey), criteria.value + "%")
-            SearchOperation.ENDS_WITH -> return builder.like(nestedRoot.get(criteriaKey), "%" + criteria.value)
-            SearchOperation.CONTAINS -> return builder.like(builder.lower(nestedRoot.get<String>(criteriaKey).`as`(String::class.java)), "%" + criteria.value + "%")
-            SearchOperation.DOESNT_START_WITH -> return builder.notLike(nestedRoot.get(criteriaKey), criteria.value + "%")
-            SearchOperation.DOESNT_END_WITH -> return builder.notLike(nestedRoot.get(criteriaKey), "%" + criteria.value)
-            SearchOperation.DOESNT_CONTAIN -> return builder.notLike(builder.lower(nestedRoot.get<String>(criteriaKey).`as`(String::class.java)), "%" + criteria.value + "%")
-            else -> return null
+        val fieldClass = nestedRoot.get<Any>(criteriaKey).javaType.kotlin
+        val strategy = ParsingStrategy.getStrategy(fieldClass)
+        val value: Any?
```

Comment:
Is it possible to create a type to specify handled types?

Reply:
I did not find a solution for this problem. 
Kotlin seems to currently have no solution for this : https://discuss.kotlinlang.org/t/union-types/77/67